cdap-ui/app/features/tracker/controllers/tags-ctrl.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+class TrackerTagsController{
+  constructor($state, $scope, myTrackerApi, $uibModal) {
+    this.$state = $state;
+    this.$scope = $scope;
+    this.myTrackerApi = myTrackerApi;
+    this.$uibModal = $uibModal;
+
+    this.tagList = {
+      preferred: 0,
```

Comment:
Any reason why we need a separate count for preferred tags? Could it be `this.tagList.preferredTags.length`?